### PR TITLE
Rename some customer cases test names in order to improve readability

### PR DESCRIPTION
**Pull request by pjgg**

### CODE REVIEW

File: spring/spring-data/src/test/java/io/quarkus/ts/spring/data/primitivetypes/BookResourceIT.java [hunk 1]
```diff
@@ -82,7 +82,7 @@ void testJpaFieldsMapping() {
```
Comment (rsvoboda) [inline spring/spring-data/src/test/java/io/quarkus/ts/spring/data/primitivetypes/BookResourceIT.java:82]: `BookResourceIT.java` name is not great. Can we have more descriptive class name about what it is testing and not what endpoint it is using for testing?

**Review by rsvoboda — CHANGES_REQUESTED** [review]

File: spring/spring-data/src/test/java/io/quarkus/ts/spring/data/primitivetypes/BookResourceIT.java (hunk 1 above)
Comment (pjgg) [reply spring/spring-data/src/test/java/io/quarkus/ts/spring/data/primitivetypes/BookResourceIT.java:82]: In this case, This scenario "BookResourceIT" is checking several cases:

- https://github.com/quarkusio/quarkus/pull/13015
- https://github.com/quarkusio/quarkus/issues/13234
- QUARKUS-1521

I think that package name "io.quarkus.ts.spring.data.primitivetypes" is good enough in order to understand, what this scenario is talking about. WDYT?

Comment (rsvoboda) [reply spring/spring-data/src/test/java/io/quarkus/ts/spring/data/primitivetypes/BookResourceIT.java:82]: The coverage is around CrudRepository, PanacheRepository and primitive types, `BookResourceIT` naming doesn't indicate that.

My point is mainly for Polarion, there is no notion of maven modules, the details tracked there are basically package name and `.java` file name. So file name matters there too. 

`BookResourceIT` => `PersistenceRepositoryIT` / `SpringDataPersistenceIT` / `SpringDataBookResourceIT`

I probably like the last one (`SpringDataBookResourceIT`) the most. It indicates the resource name (easier investigation) and the first part should be enough for context for Polarion

Comment (rsvoboda) [reply spring/spring-data/src/test/java/io/quarkus/ts/spring/data/primitivetypes/BookResourceIT.java:82]: And the same naming concern applies to `CatResourceIT`

**Review by rsvoboda — CHANGES_REQUESTED** [review]
https://github.com/quarkus-qe/quarkus-test-suite/pull/492#discussion_r788845023

**Review by rsvoboda — APPROVED** [review]

**Pull request merged**